File: leo2-node/src/main/java/org/deku/leo2/node/data/sync/v1/EntityUpdateMessage.java
package org.deku.leo2.node.data.sync.v1;

import java.io.Serializable;

/**
 * Message sent by publishers prior to the actual entity payload, containing meta information
 * eg. amount of entites to expect
 * Created by masc on 18.06.15.
 */
public class EntityUpdateMessage implements Serializable {
    private static final long serialVersionUID = -8032738544698874536L;

    public static final String EOS_PROPERTY = "eos";

    private Long mAmount;

    public EntityUpdateMessage() { }

    /**
     * c'tor
     * @param amount Amount of records
     */
    public EntityUpdateMessage(Long amount) {
        mAmount = amount;
    }

    public Long getAmount() {
        return mAmount;
    }

    @Override
    public String toString() {
        return String.format("%s amount [%d]", this.getClass().getSimpleName(), this.getAmount());
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/data/sync/v1/EntityUpdateMessage.kt
package org.deku.leoz.node.data.sync.v1

import java.io.Serializable

/**
 * Message sent by publishers prior to the actual entity payload, containing meta information
 * eg. amount of entites to expect
 * @param amount Amount of entities/records to follow
 * Created by masc on 18.06.15.
 */
class EntityUpdateMessage(
        val amount: Long = 0)
: Serializable {
    override fun toString(): String {
        return String.format("%s amount [%d]", this.javaClass.simpleName, this.amount)
    }

    companion object {
        private val serialVersionUID = -8032738544698874536L

        /** Property set on last message to indicate end of stream  */
        val EOS_PROPERTY = "eos"
    }
}
